src/core/layertree/qgslayertreemodel.cpp

```diff
@@ -941,6 +941,14 @@ void QgsLayerTreeModel::connectToLayer( QgsLayerTreeLayer *nodeLayer )
     connect( vl, &QgsVectorLayer::layerModified, this, &QgsLayerTreeModel::layerNeedsUpdate, Qt::UniqueConnection );
   }
 
+  if ( layer->type() == QgsMapLayerType::MeshLayer )
+  {
+    QgsMeshLayer *ml = qobject_cast< QgsMeshLayer * >( layer );
+    connect( ml, &QgsMeshLayer::frameEditingStarted, this, &QgsLayerTreeModel::layerNeedsUpdate, Qt::UniqueConnection );
```

Comment:
Here's another case where signals in QgsMapLayer may be more appropriate. Could the vector layer editingStarted/editingStopped/layerModified signals be moved to QgsMapLayer and used in place of the mesh specific signals?